clang/lib/CIR/CodeGen/CIRGenBuilder.h

```diff
@@ -458,11 +458,21 @@ class CIRGenBuilderTy : public mlir::OpBuilder {
                                      mlir::cir::CastKind::ptr_to_int, src);
   }
 
+  // FIXME(cir): the following function does not make sense in CIR's context
+  // since we do not have "zext" operations. We should understand where its
+  // used in the original code gen so that we can create something equivalent
+  // in CIR. For now, we just add whatever cast is required.
```

Comment:
This was introduced to keep in sync LLVM codegen but so far has only been used to short-circuit the case where types match, which makes this logically a no-op before this patch. I think it's valuable (though slightly misleading) to keep the function name for the sake of comparing both codegen stories. Can you update the comment to actually reflect that instead of a TODO? Any other thoughts or suggestions?

Reply:
@bcardosolopes but at some point, we intend to detach from the original codegen right? Meaning we will replace CIRGen stuff with whatever makes more sense to CIR. This is why I marked it as a FIXME.